src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/NodeOrFilter.cs

```diff
@@ -0,0 +1,63 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.Nodes
+{
+    /// <summary>
+    /// Has information for filtering a SMO object by properties 
+    /// </summary>
+    public class NodeOrFilter : INodeFilter
+    {
+        /// <summary>
+        /// Filter values
+        /// </summary>
+        public List<NodePropertyFilter> FilterList { get; set; }
+
+        /// <summary>
+        /// Returns true if any of the filters within the FilterList apply to the type and server type.
+        /// Note: Not currently used, because the validity of the filter is essentially only possible while
+        /// constructing the filter string, for each individual filter.
+        /// </summary>
+        /// <param name="type">Type of the querier</param>
+        /// <param name="validForFlag">Server Type</param>
+        public bool CanApplyFilter(Type type, ValidForFlag validForFlag) {
+            bool canApplyFilter = false;
+            foreach (var filter in FilterList) {
+                canApplyFilter |= filter.CanApplyFilter(type, validForFlag);
+            }
+            return canApplyFilter;
+        }
+
+        /// <summary>
+        /// Creates a string representation of a node "or" filter, which is combined in the INodeFilter interface to construct the filter used in the URN to query the SQL objects.
+        /// Example of the output: ((@TableTemporalType = 1) or (@LedgerTableType = 1))
+        /// </summary>
+        /// <returns></returns>
+        public string ToPropertyFilterString(Type type, ValidForFlag validForFlag)
```

Comment:
Can you just use INodeFilter.GetPropertyFilter and have that take in the and/or value to use between filter values? The logic seems to be the same. 

(looks like you'd have to have a value for the types of brackets to use too)

I would suggest probably just making a static helper class at that point to keep all that generic logic in one place.